### PR TITLE
Hyrax parallel `commit`

### PR DESCRIPTION
**Pull request by mmagician**
## Description
hyrax before:
`commit Hyrax254/20      time:   [1.0561 s 1.0618 s 1.0686 s]`

hyrax with parallel commits per column:
`commit Hyrax254/20      time:   [534.62 ms 534.97 ms 535.31 ms]`

Benched on 16 cores / 32 threads. The effect is smaller than what we saw for other PCS (was it 70% reduction for Ligero IIRC?), but still quite significant compared to the non-parallel version.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by mmagician** [comment]
@burdges I know you're experienced with `rand`, maybe you could have a peek and see if what I'm doing in this PR is any safe/efficient/correct?

File: poly-commit/src/hyrax/mod.rs [hunk 1]
```diff
@@ -272,15 +272,16 @@ impl<G: AffineRepr, P: MultilinearExtension<G::ScalarField>>
...
+            let (row_coms, com_rands): (Vec<_>, Vec<_>) = cfg_iter!(m)
                 .map(|row| {
+                    #[cfg(not(feature = "parallel"))]
                     let (c, r) = Self::pedersen_commit(ck, row, None, Some(rng_inner));
-                    // Storing the randomness used in the commitment
-                    com_rands.push(r);
-                    c
+                    #[cfg(feature = "parallel")]
+                    let (c, r) =
+                        Self::pedersen_commit(ck, row, None, Some(&mut rand::thread_rng()));
```
Comment (Antonio95) [inline poly-commit/src/hyrax/mod.rs:281]: Is this `thread_rng` strong enough? (e.g. stronger than the `test_rng`s we normally use for testing). If so, I'm happy with all the code

Comment (mmagician) [reply poly-commit/src/hyrax/mod.rs:281]: It's definitely stronger than test_rng :D

Comment (Antonio95) [reply poly-commit/src/hyrax/mod.rs:281]: Seems solid: https://rust-random.github.io/rand/rand/fn.thread_rng.html

Merge away!

**Pull request merged**